src/rvs_blas.cpp

```diff
@@ -916,86 +942,115 @@ bool rvs_blas::run_blas_gemm(void) {
  */
 void rvs_blas::generate_random_matrix_data(void) {
 
-  size_t i;
   if (!is_error) {
-    uint64_t nextr = (uint64_t) time(NULL);
 
-    //SGEMM (float fp32_r)
-    if(ops_type == "sgemm") {
+    if("hiprand" == matrix_init) {
 
-      for (i = 0; i < size_a; ++i)
-        ha[i] = fast_pseudo_rand(&nextr, i);
+      if(ops_type == "dgemm") {
 
-      for (i = 0; i < size_b; ++i)
-        hb[i] = fast_pseudo_rand(&nextr, i);
+        if(hiprandGenerateUniformDouble(hiprand_generator, ddbla, size_a) != HIPRAND_STATUS_SUCCESS) {
+          std::cout << "\n hiprandGenerateUniformDouble() failed !!!" << "\n";
+          return;
```

Comment:
in case of any of these failures again return value not checked or used in caller and it proceeds for operations as if no error.(in iet and gst)

Reply:
error set is function calls fails.